code/controllers/ShortcodableController.php

```diff
@@ -6,19 +6,17 @@
  **/
 class ShortcodableController extends LeftAndMain
 {
-    /**
-     * @var string
-     */
-    const URLSegment = 'ShortcodableController';
+    private static $url_segment = 'shortcodes';
+    private static $menu_title = 'Shortcodes';
+    private static $required_permission_codes = 'CMS_ACCESS_AssetAdmin';
```

Comment:
@sheadawson all works fine for Admin role, problem is if you are logged as Content Author with simple Permissions. Pop-up modal for adding shortcodes is blank in this case and using `$required_permission_codes` resolves issue .

After more testing I would suggest using permission `CMS_ACCESS_CMSMain (Access to 'Pages' section)`.
It is used in `CMSPageAddController` and it seems logical to continue with same security level for shortcodes.

What do you think?